### PR TITLE
refactor: Explode index & App components

### PR DESCRIPTION
**Pull request by Merkur39**
For reasons of maintainability and future evolution of the router

### CODE REVIEW

**Review by JF-Cozy — APPROVED** [review]
💥

File: src/components/OnboardedGuardedRoute.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+'use strict'
```
Comment (trollepierre) [inline src/components/OnboardedGuardedRoute.spec.jsx:1]: not needed ?

File: src/targets/browser/index.jsx [hunk 2]
```diff
@@ -5,63 +5,18 @@ import '../../styles/index.styl'
...
-                </WebviewIntentProvider>
-              </BreakpointsProvider>
-            </MuiCozyTheme>
-          </ScannerI18nProvider>
-        </I18n>
-      </CozyProvider>
-    </StylesProvider>,
+    <AppProviders {...rest}>
+      <App />
+    </AppProviders>,
```
Comment (trollepierre) [inline src/targets/browser/index.jsx:19]: now we can easily test this component (if mocking AppProviders) 🙏

File: src/components/AppRouter.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    component: FilesViewerWithQuery
+  },
+  {
+    path: '/onboarding',
+    component: Onboarding
+  }
+]
+
+export const AppRouter = () => {
+  return (
```
Comment (trollepierre) [inline src/components/AppRouter.jsx:30]: remove unnecessary return here 🙏

File: src/components/AppRouter.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    path: '/file/:fileId',
+    component: FilesViewerWithQuery
+  },
+  {
+    path: '/onboarding',
+    component: Onboarding
+  }
+]
+
+export const AppRouter = () => {
```
Comment (trollepierre) [inline src/components/AppRouter.jsx:29]: it's time to test AppRouter (you can remove one assertion) :  

```
import React from 'react'
import { render } from '@testing-library/react'
import { AppRouter } from 'src/components/AppRouter'

/* eslint-disable react/display-name */
jest.mock('src/components/OnboardedGuardedRoute', () => ({ component }) => (
  <div data-testid="OnboardedGuardedRoute">{component()}</div>
))
jest.mock('src/components/Home/Home', () => () => <div data-testid="Home" />)
jest.mock('src/components/Onboarding/Onboarding', () => () => (
  <div data-testid="Onboarding" />
))
jest.mock('src/components/Papers/PapersListWrapper', () => () => (
  <div data-testid="PapersListWrapper" />
))
jest.mock('src/components/Viewer/FileViewerWithQuery', () => () => (
  <div data-testid="FileViewerWithQuery" />
))
jest.mock('react-router-dom', () => ({
  HashRouter: ({ children }) => <div data-testid="HashRouter">{children}</div>,
  Redirect: ({ children }) => <div data-testid="Redirect">{children}</div>,
  Switch: ({ children }) => <div data-testid="Switch">{children}</div>
}))
/* eslint-enable react/display-name */

describe('AppRouter', () => {
  it('should render home, onboarding, papersListWrapper, fileViewerWithQuery', () => {
    // When
    const { container, queryByTestId } = render(<AppRouter />)

    // Then
    expect(queryByTestId('Home')).toBeTruthy()
    expect(queryByTestId('Onboarding')).toBeTruthy()
    expect(queryByTestId('PapersListWrapper')).toBeTruthy()
    expect(queryByTestId('FileViewerWithQuery')).toBeTruthy()
    expect(container).toMatchInlineSnapshot(`
      <div>
        <div
          data-testid="HashRouter"
        >
          <div
            data-testid="Switch"
          >
            <div
              data-testid="OnboardedGuardedRoute"
            >
              <div
                data-testid="Home"
              />
            </div>
            <div
              data-testid="OnboardedGuardedRoute"
            >
              <div
                data-testid="PapersListWrapper"
              />
            </div>
            <div
              data-testid="OnboardedGuardedRoute"
            >
              <div
                data-testid="FileViewerWithQuery"
              />
            </div>
            <div
              data-testid="OnboardedGuardedRoute"
            >
              <div
                data-testid="Onboarding"
              />
            </div>
            <div
              data-testid="Redirect"
            />
          </div>
        </div>
      </div>
    `)
  })
})
```

**Review by trollepierre — CHANGES_REQUESTED** [review]

File: src/components/App.jsx [hunk 5]
```diff
@@ -1,132 +1,35 @@
...
+        />
+      ) : (
+        <AppRouter />
+      )}
+      {isStepperDialogOpen && (
+        <FormDataProvider>
+          <StepperDialogWrapper />
+        </FormDataProvider>
+      )}
+    </AppLayout>
```
Comment (trollepierre) [inline src/components/App.jsx:32]: now I think we are able to test this component 🙏

File: src/components/AppLayout.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          {children}
+        </Content>
+
+        <RealTimeQueries doctype="io.cozy.files" />
+        <RealTimeQueries doctype="io.cozy.mespapiers.settings" />
+        <Alerter t={t} />
+        <ModalStack />
+      </Main>
+      <IconSprite />
+    </Layout>
```
Comment (trollepierre) [inline src/components/AppLayout.jsx:55]: nit: personally, I would have left a big snapshot by mocking all the imported component.

File: src/components/AppProviders.jsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+              <BreakpointsProvider>
+                <WebviewIntentProvider>
+                  <LocalProviders>{children}</LocalProviders>
+                </WebviewIntentProvider>
+              </BreakpointsProvider>
+            </MuiCozyTheme>
+          </ScannerI18nProvider>
+        </I18n>
+      </CozyProvider>
+    </StylesProvider>
```
Comment (trollepierre) [inline src/components/AppProviders.jsx:58]: nit: personally, I would have left a big snapshot by mocking all the imported component.

**Review by trollepierre — COMMENTED** [review]
Thanks for the refactor 🎉 It improves a lot the code readability :-)

File: src/components/AppRouter.jsx (hunk 4 above)
Comment (Merkur39) [reply src/components/AppRouter.jsx:29]: Thanks for this nice example 👍

**Review by trollepierre — APPROVED** [review]
Wonderful! I really appreciated our pair programming session. Tests are much more resistant 💪

**Pull request merged**